SingularityService/src/main/java/com/hubspot/singularity/data/RequestManager.java

```diff
@@ -282,6 +287,24 @@ public SingularityDeleteResult markDeleted(SingularityRequest request, long time
   }
 
   public List<SingularityRequestWithState> getRequests(Collection<String> requestIds) {
+    return getRequests(requestIds, false);
+  }
+
+  public List<SingularityRequestWithState> getRequests(Collection<String> requestIds, boolean useWebCache) {
+    if (leaderCache.active()) {
+      return leaderCache.getRequests().stream().filter((r) -> requestIds.contains(r.getRequest().getId())).collect(Collectors.toList());
+    }
+
+    if (useWebCache) {
+      if (webCache.useCachedRequests()) {
+        return webCache.getRequests().stream().filter((r) -> requestIds.contains(r.getRequest().getId())).collect(Collectors.toList());
+      } else {
+        List<SingularityRequestWithState> requests = getRequests(true);
+        webCache.cacheRequests(requests);
```

Comment:
I was debating if it was helpful to repopulate the cache here. This fetches all the requests then the returned values are the filtered ones. This particular method is called a bunch by the authorization helper to fetch requests allowed to be viewed by a user. We would previously only fetch the relevant ones from zk (still a large list on average), but that's not enough to repopulate the cache since it isn't the full list